src/main/java/com/sda/controller/CostumeController.java

```diff
@@ -33,9 +31,9 @@ public ModelAndView findAllCostumes(@PathVariable(required = false) Long id) {
     }
 
     @GetMapping("/costumes")
-    public ModelAndView showAll() {
+    public ModelAndView showAll(Model model, @RequestParam(defaultValue="") String name) {
```

Comment:
Myślę że lepiej byłoby rozbić to na dwie różne metody, w jednej w którym query parametr name jest required (showByName) oraz drugi bez wskazania query parametrów (showAll)